compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/packaging/converters/FileSystemSourceInput.java

```diff
@@ -15,15 +16,31 @@
 public class FileSystemSourceInput implements CompilerInput {
 
     private final Path path;
+    private Path packageRoot;
 
     public FileSystemSourceInput(Path path) {
         this.path = path;
     }
 
+    public FileSystemSourceInput(Path filePath, Path packageRoot) {
+        this.path = filePath;
+        this.packageRoot = packageRoot;
+    }
+
     @Override
     public String getEntryName() {
-        Path fileName = path.getFileName();
-        return fileName == null ? path.toString() : fileName.toString();
+        // For non-default packages, we need to return the file path relative to the package root
```

Comment:
Better to add full stops at the end of comments.

Reply:
Commit b198293bc2026eb33a5db4c4d28522fe5e3abe1a addresses this suggestion.